### PR TITLE
Extract `deflicker` plugin and split `Screen` state

### PR DESCRIPTION
**Pull request by MiniaczQ**

### CODE REVIEW

**Review by janhohenheim — APPROVED** [review]

File: src/core/deflicker.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+#[derive(States, Debug, Hash, PartialEq, Eq, Clone, Default)]
+enum Deflicker {
+    #[default]
+    Running,
+    Done,
+}
+
+fn make_window_invisible(mut window: Query<&mut Window>) {
```
Comment (benfrankel) [inline src/core/deflicker.rs:23]:
```suggestion
fn hide_window(mut window: Query<&mut Window>) {
```

File: src/core/deflicker.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    window.single_mut().visible = cfg!(target_os = "windows");
+}
+
+fn finish_deflicker(frames: Res<FrameCount>, mut next_deflicker: ResMut<NextState<Deflicker>>) {
+    if frames.0 >= 3 {
+        next_deflicker.set(Deflicker::Done)
+    }
+}
+
+fn make_window_visible(mut window: Query<&mut Window>) {
```
Comment (benfrankel) [inline src/core/deflicker.rs:35]:
```suggestion
fn show_window(mut window: Query<&mut Window>) {
```

**Review by benfrankel — APPROVED** [review]
Feel free to ignore the system rename suggestions.

File: src/core/mod.rs [hunk 3]
```diff
@@ -2,9 +2,10 @@
 
 mod asset;
 mod camera;
+mod deflicker;
 #[cfg(feature = "dev")]
 mod dev;
-mod window;
+pub mod window;
```
Comment (janhohenheim) [inline src/core/mod.rs:8]:
```suggestion
pub(crate) mod window;
```

File: src/core/window.rs [hunk 4]
```diff
@@ -13,3 +14,10 @@ pub(super) fn plugin(app: &mut App) {
         ..default()
     });
 }
+
+#[derive(States, Debug, Hash, PartialEq, Eq, Clone, Default)]
+pub enum WindowState {
```
Comment (janhohenheim) [inline src/core/window.rs:19]: Does this also need to go into the `dev` debug prints?

**Review by janhohenheim — CHANGES_REQUESTED** [review]

File: src/core/window.rs (hunk 4 above)
Comment (janhohenheim) [reply src/core/window.rs]: (This applies to `BootingState`)

**Review by benfrankel — APPROVED** [review]

**Review by janhohenheim — APPROVED** [review]

**Review by benfrankel — APPROVED** [review]
Will be very easy to move `booting.rs` back into `window.rs` later if it turns out to be possible while decoupling the asset loading stuff. So LGTM.

**Pull request merged**